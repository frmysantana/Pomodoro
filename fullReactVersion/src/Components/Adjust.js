import React from 'react';

//Used class component for the life-cycle method
export default class Adjust extends React.Component {
    changeTime = (e) => {
      /* Invokes the root components' changeTime function with the correct
        parameters needed for the specific click event that activated it.

        Inputs: the click event from either the '+' or '-' buttons.
        Outputs: None.
       */
      if (e.target.id === 'add') {
        console.log()
        this.props.changeTime('add', 'maximum', 59.59, 29.59);
      } else {
        this.props.changeTime('sub', 'minimum', 5.00, 1.00);
      }
    }

    changeUnit = (e) => {
      this.props.changeUnit(e.target.id);
    }
    
    // So user knows that by default, they would alter the minutes portion of a 
    // timer
    componentDidMount() {
      // So that this lifecycle method doesn't interfere with the test files.
      if (document.querySelector('#min')) {
        document.querySelector('#min').checked = true;
      }
    }

    render() {
      return (
        <div id="Adjust">
          <div className="choice">
            <label htmlFor="minutes">Minutes</label>
            <input type="radio" name="unit"
              id="min" onClick = {this.changeUnit}
            />
          </div>

          <div className="choice">
            <label htmlFor="seconds">Seconds</label>
            <input type="radio" name="unit" 
              id="sec" onClick = {this.changeUnit}
            />
          </div>
          
          <div id="buttons">
            <button onClick={this.changeTime} className="js-adjust-time" id="add">+</button>
            <button onClick={this.changeTime} className="js-adjust-time" id="sub">-</button>
          </div>
        </div>
      );
    }
}
