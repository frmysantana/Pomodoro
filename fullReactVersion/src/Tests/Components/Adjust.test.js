import React from 'react';
import Adjust from '../../Components/Adjust';
import { shallow } from 'enzyme';

test('Should render component correctly.', () => {
  const wrapper = shallow(<Adjust />);
  expect(wrapper).toMatchSnapshot();
});