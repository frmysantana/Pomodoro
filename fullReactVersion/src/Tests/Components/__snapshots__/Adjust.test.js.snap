// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component correctly. 1`] = `
<div
  id="Adjust"
>
  <div
    className="choice"
  >
    <label
      htmlFor="minutes"
    >
      Minutes
    </label>
    <input
      id="min"
      name="unit"
      onClick={[Function]}
      type="radio"
    />
  </div>
  <div
    className="choice"
  >
    <label
      htmlFor="seconds"
    >
      Seconds
    </label>
    <input
      id="sec"
      name="unit"
      onClick={[Function]}
      type="radio"
    />
  </div>
  <div
    id="buttons"
  >
    <button
      className="js-adjust-time"
      id="add"
      onClick={[Function]}
    >
      +
    </button>
    <button
      className="js-adjust-time"
      id="sub"
      onClick={[Function]}
    >
      -
    </button>
  </div>
</div>
`;
